### PR TITLE
Better Package support

### PR DESCRIPTION
**Pull request by victor-sarda**
This PR adds better native package support to PackageGenerator.
The main changes are: 

- Added a `GeneratePackages` command to generate more than one package at once
- Replaced the mention of "modules" with "packages"
- Added support for multiple products (library, executables, plugins)
- Added support for executable targets
- Added support for platforms
- Test targets are now part of the `target` array and `Target` now has a `targetType` property (closer to what SPM does)
- Replaced `path` with `sourcesPath` in `Target`
- Replaced `hasResources` with `resourcesPath` in `Target`
- Added support for more properties in `Target` (`exclude`, `cxxSettings`, `linkerSettings`, `publicHeaderPath`, `plugins`)
- Remote dependencies in the manifest files don't need the version & url placeholder anymore
- Added test for different scenarios

### CODE REVIEW

File: .swiftpm/xcode/xcshareddata/xcschemes/GeneratePackage.xcscheme [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+         </TestableReference>
+      </Testables>
+   </TestAction>
+   <LaunchAction
+      buildConfiguration = "Debug"
+      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
+      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      launchStyle = "0"
+      useCustomWorkingDirectory = "YES"
+      customWorkingDirectory = "/Users/victor.sarda/Developer/PackageGenerator"
```
Comment (albertodebortoli) [inline .swiftpm/xcode/xcshareddata/xcschemes/GeneratePackage.xcscheme]: best unticking using the custom working directory + removing this value

File: .swiftpm/xcode/xcshareddata/xcschemes/GeneratePackages.xcscheme [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+         </TestableReference>
+      </Testables>
+   </TestAction>
+   <LaunchAction
+      buildConfiguration = "Debug"
+      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
+      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      launchStyle = "0"
+      useCustomWorkingDirectory = "YES"
+      customWorkingDirectory = "/Users/victor.sarda/Developer/PackageGenerator"
```
Comment (albertodebortoli) [inline .swiftpm/xcode/xcshareddata/xcschemes/GeneratePackages.xcscheme]: same here

File: Example/Packages/Example/Package.swift [hunk 3]
```diff
@@ -39,23 +42,26 @@ let package = Package(
...
             resources: [
                 .process("Resources")
             ],
-            plugins: []
+            plugins: [
+            ]
         ),
         .binaryTarget(
             name: "GoogleMaps",
             path: "../../vendor/GoogleMaps-7.1.0/GoogleMaps.xcframework"
```
Comment (albertodebortoli) [inline Example/Packages/Example/Package.swift]: is this intended to be here?

**Review by albertodebortoli — APPROVED** [review]
Amazing work! 👏

File: Example/Packages/Example/Package.swift (hunk 3 above)
Comment (victor-sarda) [reply Example/Packages/Example/Package.swift]: mmh I'm not sure I changed it, I'll make it more generic though!

**Review by albertodebortoli — APPROVED** [review]

**Pull request merged**